value.go

```diff
@@ -253,11 +250,41 @@ func newExceptionValue(s *C.mrb_state) *Exception {
 		panic("exception value init without exception")
 	}
 
+	arenaIndex := C.mrb_gc_arena_save(s)
+
 	// Convert the RObject* to an mrb_value
 	value := C.mrb_obj_value(unsafe.Pointer(s.exc))
 
+	// Retrieve and convert backtrace to []string (avoiding reflection in Decode)
+	var backtrace []string
+	mrbBacktrace := newValue(s, C.mrb_exc_backtrace(s, value)).Array()
+	for i := 0; i < mrbBacktrace.Len(); i++ {
+		ln, _ := mrbBacktrace.Get(i)
+		backtrace = append(backtrace, ln.String())
+	}
+
+	// Extract file + line from first backtrace line
+	file := "Unknown"
+	line := 0
+	if len(backtrace) > 0 {
+		fileAndLine := strings.Split(backtrace[0], ":")
+		if len(fileAndLine) >= 2 {
+			file = fileAndLine[0]
+			line, _ = strconv.Atoi(fileAndLine[1])
```

Comment:
Indeed, fair point. I took the approach that if an error occurs during error parsing, we probably want to return as much of the original error as possible rather than pollute with error parsing errors.

This said, the backtrace copy was written to the current implementation of the Array type which will only ever return nil for err (something I pointed out in another PR but @mitchellh wanted to keep in case that should ever change) so, as it stands its reasonably safe (though I appreciate it has the opportunity to bite in the future).

Similarly Atoi will return the zero value for int in the case of an error which seems like reasonable behaviour although on further consideration may be indicative that mruby changed the string format...